compiler/semtypinst.nim

```diff
@@ -462,6 +462,14 @@ proc propagateFieldFlags(t: PType, n: PNode) =
       propagateFieldFlags(t, son)
   else: discard
 
+template genericParamsToString(t: untyped): untyped =
+  var genericParams = "["
+  for i in countup(0, sonsLen(t)-1-ord(t.kind != tyGenericInvocation)):
```

Comment:
araq may complain about spacing around operators (eg `a-b` => `a - b`)
we really need `nimpretty --diff` (analog of `git clang-format`), cf https://github.com/nim-lang/Nim/issues/8458
this would eliminate such concerns, while avoiding to create large diffs

Reply:
This was copied directly from here with no modification:
https://github.com/nim-lang/Nim/blob/05c52ff34f996bb44425bcc17ae3ae60ed896bef/compiler/types.nim#L461

If it's wrong here, it's wrong where I got it from. Araq obviously let that version pass.